src/model/types.rs

```diff
@@ -0,0 +1,82 @@
+use std::str::FromStr;
+
+/// Enumerates the two possible stone types
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Stone {
+    /// Stone placed by the current player.
+    ///
+    /// Defined as numerical value 1 in the piskwork protocol
+    Ally = 1,
+
+    /// Stone placed by the opponent player.
+    ///
+    /// Defined as numerical value 2 in the piskwork protocol
+    Opponent = 2,
+}
+
+/// Enumerates all possible states of a Gomoku cell
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum CellContent {
+    /// No stone placed in this cell.
+    ///
+    /// Defined as numerical value 0 in the piskwork protocol
+    Empty = 0,
+
+    /// Cell contains a stone placed by the current player.
+    ///
+    /// See [`Stone::Ally`]
+    Ally = Stone::Ally as isize,
+
+    /// Cell contains a stone placed by the opponent player.
+    ///
+    /// See [`Stone::Opponent`]
+    Opponent = Stone::Opponent as isize,
```

Comment:
I suppose there is a reason for `as isize` ?

Reply:
This is actually required by the compiler. Unlike C, Rust doesn't allow for implicit conversion between enumerators and integers.
However it does map enumerators to integer constants, just like C.
Like C once again, by default enumerators are given values in ascending order of definition, starting from zero.
If we want to explicitly set a value to an enumerator, we must give it a value of the `enum`'s storage type, which is `isize` by default (signed integer type of as many bytes as pointer type).

I was then afraid this meant that storing our enums in memory would take 8 bytes per value, since pointer are 8 bytes long on x64, but it seems like the enums' default storage type is `isize` only at compile time, and then adjusted to fit on the fewest bytes possible, as you can see in the tests calling `std::mem::size_of`.

All of that to say that the reason why we need to specify `as isize` is because explicit values of enumerators should be given in the enum's internal storage type, which is by default `isize` at compile type, which we need to explicitly cast from the other enumerator.

Reply:
Very well 🫡